HW1/P6/P6.py

```diff
@@ -0,0 +1,72 @@
+import findspark
+findspark.init()
+import pyspark
+import time
+import re
+from random import random
+from bisect import bisect
+start = time.time()
+sc = pyspark.SparkContext(appName='test')
+
+def mapFunc(word):
+	wordList = []
+	word = word.encode('utf-8')
+	wordList = re.split(" ", word)
+	return wordList
+
+
+def filterFunc(word):
+	if len(word) == 0:
+		return False
+	return (not word[:-1].isupper()) if word[-1] == '.' else  not word.isupper() and not word.isdigit()
+
+textRDD = sc.textFile('pg100.txt',8)
+rawRDD = textRDD.flatMap(mapFunc)
+wordsRDD = rawRDD.filter(filterFunc) # Now wordsRDD contains all the desired words
+
+firstWords, secondWords, thirdWords = wordsRDD.zipWithIndex().map(lambda (k, v): (v, k)), wordsRDD.zipWithIndex().map(lambda (k, v):(v-1,k)).filter(lambda (k, v): k >= 0), wordsRDD.zipWithIndex().map(lambda (k, v): (v-2, k)).filter(lambda (k, v):k >= 0)
+threeWordSet = firstWords.join(secondWords).join(thirdWords).map(lambda (k, v): v)
+
+zipSameWord = threeWordSet.groupByKey().map(lambda (k, v): (k, (list(v))))
+zipSameWordDict = zipSameWord.map(lambda (k, v): (k,dict((x,v.count(x)) for x in v)))
+keyWord = ('Now', 'is')
+resList = zipSameWordDict.map(lambda x: x).lookup(keyWord)
+res = resList[0].items()
+
+def weighted_choice(choices):
+    values, weights = zip(*choices)
+    total = 0
+    cum_weights = []
+    for w in weights:
+        total += w
+        cum_weights.append(total)
+    x = random() * total
+    i = bisect(cum_weights, x)
```

Comment:
Alternatively, you can use numpy.random.choice which draws an element from an array based on weights.